### ISSUE TITLE
External input in lambda layer - which will be updated 

### ISSUE THREAD

**Issue by cskostas87** [fchollet/keras#5508]
- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

Hello, i have a triplet network like below and i want in lambda layer to have a global table will be used in calculations and also will be updated. i have also a custom objective for triplet loss.
how can i achieve this?
another way is to compute the gradients apart from train procedure and only doing a backpropagation.

specifically, my problem is:
table1 = somedata;
for each epoch:
 for each batch:
     outputs = model.predict ()
     table1 = UpdateValues(table1, ouputs)
     loss = CalcLoss (outputs, table1) 
     gradients = CalcGradients (loss,  table1)
     backward pass(gradients)
 end
end
![model](https://cloud.githubusercontent.com/assets/25438904/23313434/63c98918-fac6-11e6-8d58-55467746260b.png)
![base_network](https://cloud.githubusercontent.com/assets/25438904/23313438/660a7778-fac6-11e6-8e2e-43ab07ccad0b.png)

**Comment by unrealwill** [comment]
Hello,

Have you tried theano with your generators ?
Have you tried to use train_on_batch, instead of fit ?

You can always go down a level and do it manually either in tensorflow or theano.

**Comment by cskostas87** [comment]
i have already used train_on_batch, i am working only with tensorFlow , i haven't tried theano yet. and i know a little about theano lib.
